From dd17862a9e2a10ac681c608fde93a91c8e2879cb Mon Sep 17 00:00:00 2001
From: Marty Jones <mj8263788@gmail.com>
Date: Fri, 6 Jun 2025 19:59:32 -0400
Subject: [PATCH 1/2] linux-firmware: realtek: add firmware for
 RTL8814AE/RTL8814AU

This adds firmware support for the RTL8814AE/RTL8814AU WiFi adapters.

Signed-off-by: Marty Jones <mj8263788@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/19052
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>
---
 package/firmware/linux-firmware/realtek.mk | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/package/firmware/linux-firmware/realtek.mk b/package/firmware/linux-firmware/realtek.mk
index 749d93be3e..ccbc4b1870 100644
--- a/package/firmware/linux-firmware/realtek.mk
+++ b/package/firmware/linux-firmware/realtek.mk
@@ -147,6 +147,13 @@ define Package/rtl8812a-firmware/install
 endef
 $(eval $(call BuildPackage,rtl8812a-firmware))
 
+Package/rtl8814a-firmware = $(call Package/firmware-default,RealTek RTL8814AU firmware)
+define Package/rtl8814a-firmware/install
+	$(INSTALL_DIR) $(1)/lib/firmware/rtw88
+	$(INSTALL_DATA) $(PKG_BUILD_DIR)/rtw88/rtw8814a_fw.bin $(1)/lib/firmware/rtw88
+endef
+$(eval $(call BuildPackage,rtl8814a-firmware))
+
 Package/rtl8821a-firmware = $(call Package/firmware-default,RealTek RTL8821AU firmware)
 define Package/rtl8821a-firmware/install
 	$(INSTALL_DIR) $(1)/lib/firmware/rtw88
-- 
2.43.0

